src/scripts/main.js

```diff
@@ -1,5 +1,33 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  const wallSize = wall.getBoundingClientRect();
+  const spiderWidth = spider.offsetWidth;
+  const spiderHeight = spider.offsetHeight;
+
+  if (
+    e.clientX >= wallSize.left &&
+    e.clientX <= wallSize.right &&
+    e.clientY >= wallSize.top &&
+    e.clientY <= wallSize.bottom
+  ) {
+    let newLeftCoords = e.clientX - wallSize.left - spiderWidth / 2 - 10;
+    let newTopCoords = e.clientY - wallSize.top - spiderHeight / 2 - 10;
```

Comment:
The subtraction of 10 from both `newLeftCoords` and `newTopCoords` seems arbitrary and may not be necessary. This could cause the spider to be positioned incorrectly. Consider reviewing whether this offset is needed for your specific use case.